libs/server/Resp/RespServerSessionSlotVerify.cs

```diff
@@ -30,24 +30,40 @@ bool NetworkSingleKeySlotVerify(ReadOnlySpan<byte> key, bool readOnly)
         bool NetworkKeyArraySlotVerify(Span<ArgSlice> keys, bool readOnly, int count = -1)
             => clusterSession != null && clusterSession.NetworkKeyArraySlotVerify(keys, readOnly, SessionAsking, ref dcurr, ref dend, count);
 
-        /// <summary>
-        /// Verify if the corresponding command can be served given the status of the slot associated with the parsed keys.
-        /// </summary>
-        /// <param name="readOnly"></param>
-        /// <param name="firstKey"></param>
-        /// <param name="lastKey"></param>
-        /// <param name="step"></param>
-        /// <returns></returns>
-        bool NetworkMultiKeySlotVerify(bool readOnly = false, int firstKey = 0, int lastKey = -1, int step = 1)
+        bool CanServeSlot(RespCommand cmd)
         {
+            // If cluster is disable all commands
             if (clusterSession == null)
+                return true;
+
+            // Verify slot for command if it falls into data command category
+            if (!cmd.IsDataCommand())
+                return true;
+
+            cmd = cmd.NormalizeForACLs();
+            if (!RespCommandsInfo.TryFastGetRespCommandInfo(cmd, out var commandInfo))
+                // This only happens if we failed to parse the json file
                 return false;
-            csvi.readOnly = readOnly;
+
+            // The provided command is not a data command so we can serve without any slot restrictions
+            if (commandInfo == null)
+                return true;
+
+            csvi.readOnly = cmd.IsReadOnly();
             csvi.sessionAsking = SessionAsking;
-            csvi.firstKey = firstKey;
-            csvi.lastKey = lastKey < 0 ? parseState.count + 1 + lastKey : lastKey;
-            csvi.step = step;
-            return clusterSession.NetworkMultiKeySlotVerify(ref parseState, ref csvi, ref dcurr, ref dend);
+            csvi.firstKey = cmd switch
+            {
+                RespCommand.ZDIFF => 1, // ZDIFF first key comes after keyCount parameter
```

Comment:
This info of first key and last key can come from the command info. ZDIFF is not the only command in which the first key is in the 2nd position... We should pre-populate this information prior to calling this method.

Reply:
RespCommandsInfo.json seems to be wrong for ZDIFF, it indicates that the first key is at position zero. I assumed that the data in the json file is auto-generated so I opted out to do it at runtime. If we can manually edit I can fix the issue. Please confirm.

Reply:
I am closing this, since it no longer applies since we are using KeySpecifications